### PR TITLE
Fixes #21741: Auto-format scala code

### PR DESCRIPTION
**Pull request by fanf**
https://issues.rudder.io/issues/21741

The formating is done with the process explained in ticket comment, everything is repeatable now. 
In other branches, some more relative imports may exists, but it should be enought to add them in the sed list.

Be careful to always do the last two `mvn` command: it happens that something breaks in very funny and novel ways, like something really looking like a comment but no, it's real scala code.

### CODE REVIEW

**Comment by fanf** [comment]
PR updated with a new commit

**Comment by fanf** [comment]
PR updated with a new commit

File: webapp/sources/ldap-inventory/inventory-api/src/main/scala/com/normation/inventory/domain/DataTypes.scala [hunk 1]
```diff
@@ -53,67 +53,71 @@ import zio._
...
+final case class SoftwareEditor(val name: String) extends AnyVal
 
 sealed trait SecurityToken {
-  def key : String
+  def key: String
 }
 
 case object SecurityToken {
-  def kind(token : SecurityToken) = {
+  def kind(token: SecurityToken) =
```
Comment (ncharles) [inline webapp/sources/ldap-inventory/inventory-api/src/main/scala/com/normation/inventory/domain/DataTypes.scala:74]: i really don't like this change - with explicit { } it's easier to see where something starts and ends, and avoid errors where we thing the code is for the method, but actually outside (or the other way round)

File: webapp/sources/ldap-inventory/inventory-api/src/main/scala/com/normation/inventory/domain/DataTypes.scala [hunk 2]
```diff
@@ -53,67 +53,71 @@ import zio._
...
     // in rudder, we ensure that at list one (k,v) pair is "UID = the node id". If missing, it's an error
-    subject.find { case (k,v) => k == ParsedSecurityToken.nodeidOID } match {
-      case None        => InventoryError.SecurityToken(s"Certificate subject doesn't contain node ID in 'UID' attribute: ${formatSubject(subject)}").fail
-      case Some((k,v)) =>
-        if(v.trim.equalsIgnoreCase(nodeId.value)) {
+    subject.find { case (k, v) => k == ParsedSecurityToken.nodeidOID } match {
+      case None =>
+        InventoryError
+          .SecurityToken(s"Certificate subject doesn't contain node ID in 'UID' attribute: ${formatSubject(subject)}")
+          .fail
```
Comment (ncharles) [inline webapp/sources/ldap-inventory/inventory-api/src/main/scala/com/normation/inventory/domain/DataTypes.scala:102]: i'm not totally sure of this one: it's more readable, but it adds lines, so making skimming through the code harder.

File: webapp/sources/ldap-inventory/inventory-api/src/main/scala/com/normation/inventory/domain/DataTypes.scala [hunk 3]
```diff
@@ -122,93 +126,88 @@ object PublicKey {
...
     } else {
       s"""-----BEGIN RSA PUBLIC KEY-----
-      |${value.grouped(80).mkString("\n")}
-      |-----END RSA PUBLIC KEY-----""".stripMargin
+         |${value.grouped(80).mkString("\n")}
+         |-----END RSA PUBLIC KEY-----""".stripMargin
     }
-  }
-  def publicKey : IOResult[java.security.PublicKey] = {
+  def publicKey: IOResult[java.security.PublicKey] =
```
Comment (ncharles) [inline webapp/sources/ldap-inventory/inventory-api/src/main/scala/com/normation/inventory/domain/DataTypes.scala:145]: ok, i really hate this  { removal

File: webapp/sources/ldap-inventory/inventory-api/src/main/scala/com/normation/inventory/domain/DataTypes.scala [hunk 4]
```diff
@@ -122,93 +126,88 @@ object PublicKey {
...
-                }
-      res    <- obj match {
-                  case a : X509CertificateHolder =>
-                    a.succeed
-                  case _ => InventoryError.Crypto(s"Key '${key}' cannot be parsed as a valid certificate").fail
-                }
+      obj <- IO.effect(reader.readObject()).mapError { e =>
+               InventoryError.CryptoEx(s"Key '${key}' cannot be parsed as a valid certificate", e)
+             }
+      res <- obj match {
```
Comment (ncharles) [inline webapp/sources/ldap-inventory/inventory-api/src/main/scala/com/normation/inventory/domain/DataTypes.scala:181]: it's a pity the <- are not aligned anymore

File: webapp/sources/ldap-inventory/inventory-api/src/main/scala/com/normation/inventory/domain/FullInventory.scala [hunk 5]
```diff
@@ -1,43 +1,43 @@
...
-*/
+ */
 
 package com.normation.inventory.domain
 
 final case class FullInventory(
-    node:NodeInventory,
-    machine:Option[MachineInventory]
+  node:    NodeInventory,
+  machine: Option[MachineInventory]
```
Comment (ncharles) [inline webapp/sources/ldap-inventory/inventory-api/src/main/scala/com/normation/inventory/domain/FullInventory.scala:42]: yes !

**Comment by fanf** [comment]
PR updated with a new commit

**Comment by fanf** [comment]
PR updated with a new commit

**Comment by fanf** [comment]
PR updated with a new commit

**Comment by fanf** [comment]
PR updated with a new commit

**Comment by fanf** [comment]
PR updated with a new commit

File: webapp/sources/ldap-inventory/inventory-api/src/main/scala/com/normation/inventory/domain/DataTypes.scala (hunk 1 above)
Comment (fanf) [reply webapp/sources/ldap-inventory/inventory-api/src/main/scala/com/normation/inventory/domain/DataTypes.scala]: it's now only for oneliners, which tends to make the code clearer (less bloated) in that case. For two lines or more, curly braces are kept.

File: webapp/sources/ldap-inventory/inventory-api/src/main/scala/com/normation/inventory/domain/DataTypes.scala (hunk 4 above)
Comment (fanf) [reply webapp/sources/ldap-inventory/inventory-api/src/main/scala/com/normation/inventory/domain/DataTypes.scala]: it's repaired, ther are now.

**Comment by fanf** [comment]
PR rebased

**Comment by fanf** [comment]
PR rebased

File: webapp/sources/ldap-inventory/inventory-api/src/main/scala/com/normation/inventory/domain/DataTypes.scala (hunk 2 above)
Comment (fanf) [reply webapp/sources/ldap-inventory/inventory-api/src/main/scala/com/normation/inventory/domain/DataTypes.scala:102]: we will need to see if after a couple weeks it's a net gain or not

**Comment by fanf** [comment]
OK, merging this PR

**Pull request merged**